File: 프로그래머스/2/42888. 오픈채팅방/오픈채팅방.py
def solution(record):
    answer = []
    id = {}
    
    for i in record:
        input_data = i.split()
        if input_data[1] not in id:
            id[input_data[1]] = input_data[2]
        else:
            if input_data[0] == "Change" or input_data[0] == "Enter":
                id[input_data[1]] = input_data[2]
                
    for i in record:
        input_data = i.split()
        if input_data[0] == "Enter":
            answer.append(id[input_data[1]] + "님이 들어왔습니다.")
        elif input_data[0] == "Leave":
            answer.append(id[input_data[1]] + "님이 나갔습니다.")

    return answer

File: 프로그래머스/2/42888. 오픈채팅방/오픈채팅방.js
function solution(record) {
    const users = {}
    const answer = []
    
    for (const item of record) {
        const [cmd, id, name] = item.split(' ')
        if (cmd !== 'Leave') {
            users[id] = name        
        }
    }
    
    for (const item of record) {
        const [cmd, id, name] = item.split(' ')
        
        if (cmd === 'Leave') {
            answer.push(`${users[id]}님이 나갔습니다.`)
        } else if (cmd === 'Enter') {
            answer.push(`${users[id]}님이 들어왔습니다.`)
        }
    }
    
    return answer
}